### PR TITLE
support multiple histogram series

### LINKED ISSUE #8151: [Feature request] histogram support multiple legend and stack
To see several type of time series in one graph, need to support this.

The graph looks like following.
histogram3

I found some of graph tools support it.
https://www.ibm.com/support/knowledgecenter/en/SSLVMB_20.0.0/com.ibm.spss.statistics.help/gpl_examples_barcharts_histogram_stack.htm

### PR DESCRIPTION
**Pull request by mtanda**
Fix https://github.com/grafana/grafana/issues/8151.

### CODE REVIEW

**Comment by mtanda** [comment]
@torkelo I add test!!

**Comment by mtanda** [comment]
I found same bug reported at https://github.com/grafana/grafana/issues/8628 .
And notice that, we shouldn't modify `data` variable by `data = [data[0]];`.

To fix the bug, I add filter before passing series data to Flot library.
https://github.com/grafana/grafana/pull/8613/files#diff-73b6d2d157efa75a5bf386d69674b2ffR371

**Comment by mtanda** [comment]
rebase to fix conflict

File: public/app/plugins/panel/graph/graph.ts [hunk 1]
```diff
@@ -305,21 +305,39 @@ coreModule.directive('grafanaGraph', function($rootScope, timeSrv, popoverSrv) {
...
+                let series = sv.series;
+                series.histogram = true;
+                if (sv.values.length > 0) {
+                  let histogram = convertValuesToHistogram(sv.values, bucketSize, normalize);
+                  series.data = histogram;
+                } else {
+                  series.data = [];
+                }
+                return series;
+              });
```
Comment (torkelo) [inline public/app/plugins/panel/graph/graph.ts]: Can the  above block of code be moved into the histogram helper file? https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/histogram.ts

File: public/app/plugins/panel/graph/graph.ts [hunk 2]
```diff
@@ -428,7 +452,9 @@ coreModule.directive('grafanaGraph', function($rootScope, timeSrv, popoverSrv) {
         let defaultTicks = panelWidth / 50;
 
         if (data.length && bucketSize) {
-          ticks = _.map(data[0].data, point => point[0]);
+          ticks = _.chain(data).map(d => {
+            return _.map(d.data, point => point[0]);
```
Comment (torkelo) [inline public/app/plugins/panel/graph/graph.ts]: why are you doing this? Also this looks pretty terrile from a performance standpoint. Never use map/each function callback looks when dealing with large value arrays, always prefer for loops (or for of loops)

Comment (torkelo) [inline public/app/plugins/panel/graph/graph.ts]: sorry about the tone :) you just really seem to like underscore/lodash chain and doing everything in a single complicated expression. Consider not using lodash chains so much and break the code up into functions and sections. It makes it easier to read and maintain.

Comment (mtanda) [inline public/app/plugins/panel/graph/graph.ts]: Sorry, I didn't know the map/each is slow. I compare map and for. Actually, map is slower than for.
https://jsperf.com/map-uniq-vs-for/1

At this line, collect all histogram x-axis value from multiple histogram data.
point[0] is histogram x-axis value, so the total count is `bucketSize x the number of time series`.

Comment (mtanda) [inline public/app/plugins/panel/graph/graph.ts]: And I should care about readability. I replace by for.
Don't mind, You are good coding guide for me :)

Comment (torkelo) [inline public/app/plugins/panel/graph/graph.ts]: that test was not very realistic. In a more realistic test (20 series with 2000 data points) the for loop was 59 times faster:
https://jsperf.com/js-map-for-uniq2/1

Comment (torkelo) [inline public/app/plugins/panel/graph/graph.ts]: actually even in your test the for loop is 12 times faster (6,749 ops per second vs 557 ops per second with the map_uniq version)

Comment (mtanda) [inline public/app/plugins/panel/graph/graph.ts]: I understand use map for bigger data is very bad performance.
I never use map for such case.

**Comment by mtanda** [comment]
`h` and `mod+h` is already assigned. use `ctrl` as the prefix key.

**Comment by mtanda** [comment]
@torkelo I think this PR is ready to merge.
Would you check this? and tell me the release target version.

**Comment by mtanda** [comment]
rebased

**Comment by alexanderzobnin** [comment]
Related to #9792

**Comment by mtanda** [comment]
Rebased.
To simplify code, I use refresh()...

**Comment by mtanda** [comment]
@alexanderzobnin Please check.

**Review by alexanderzobnin — APPROVED** [review]
Looks good, I've made minor refactor of this and going to merge.

**Comment by mtanda** [comment]
Thanks a lot!

**Pull request merged**